packages/core/src/RPA/core/windows/inspect.py

```diff
@@ -1,90 +1,106 @@
     ):
-        """Inspect Windows element under mouse pointer
+        """Inspect Windows element under mouse pointer.
 
-        :param action: which action is attached to locator
-        :param control_window: set False to not include ``Control Window`` keyword
+        :param verbose: Show exhaustive locators if `True`, otherwise just simple ones.
+        :param action: Action attached to the locator.
+        :param control_window: Include relevant ``Control Window  ...`` statement or
+            not.
+        :param recording: Where to store records.
         """
-        # TODO. Add Python syntax
-        # TODO. Add examples
+        # TODO: Python syntax support too instead of just RF.
         output = []
         with auto.UIAutomationInitializerInThread(debug=False):
             control = auto.ControlFromCursor()
             parent_control = control.GetParentControl()
-            top_level_control = control.GetTopLevelControl()
+            try:
+                top_level_control = control.GetTopLevelControl()
+            except AttributeError:
+                top_level_control = None
+                top_level_name = top_level_handle = "N/A"
+            else:
+                top_level_name = top_level_control.Name
+                top_level_handle = top_level_control.NativeWindowHandle
 
-            parent_locator = cls._get_element_key_properties(parent_control)
-            child_locator = cls._get_element_key_properties(control)
+            parent_locator = (
+                cls._get_element_key_properties(parent_control, verbose=verbose)
+                or "N/A"
+            )
+            child_locator = (
+                cls._get_element_key_properties(control, verbose=verbose) or "N/A"
+            )
             locator_path = f"{parent_locator} > {child_locator}"
-            if "name:" in child_locator:
-                locator_path = child_locator
-            elif "id:" in child_locator:
+            if "name:" in child_locator or "id:" in child_locator:
                 locator_path = child_locator
             if control_window:
-                output.append(f"Control Window  {top_level_control.Name}")
+                output.append(f"Control Window  {top_level_name}")
             if action:
                 output.append(f"{action}  {locator_path}")
             else:
                 output.append(locator_path)
+
             if recording is not None:
                 recording.append(
                     {
                         "type": "locator",
-                        "top": top_level_control.Name,
-                        "top_handle": top_level_control.NativeWindowHandle,
+                        "top": top_level_name,
+                        "top_handle": top_level_handle,
                         "x": top_level_control,
                         "locator": locator_path,
-                        "name": parent_control.Name,
-                        "control": parent_control,
```

Comment:
I think the control is needed in the inspector side.

Reply:
You're right, good catch!

Reply:
Yes, good catch!